### DISCUSSION TITLE
Suggestions/Feedback for microshift preset

### DISCUSSION THREAD

**Discussion by praveenkumar** [crc-org/crc#3437] [Show and tell]
Please share your feedback/suggestions for microshift prest.

**Comment by EmmanuelKasper** [comment]
I would appreciate if the microshift preset stays as close as possible with what https://www.redhat.com/en/technologies/device-edge uses, so that I can reproduce demo Red Hat Device Edge / reproduce Red Hat Device Edge issues  in CRC.

**Reply by praveenkumar** [reply]
@EmmanuelKasper Currently it is plain microshift without any modification from crc side except hostname and API endpoint so it should be very close.

**Reply by cfergeau** [reply]
It's also based on RHEL9, not sure what Red Hat Device Edge currently uses (RHEL8 or 9).

**Comment by josecastillolema** [comment]
The microshift preset should be allowed in machines with <9GB of memory.

For the installer (at least **crc-macos-installer**), instead of denying the installation in systems with fewer than 9 GB:
```
Red Hat OpenShift Local requires at least 9GB of memory to run
```

Maybe the installation in systems with <9GB should be allowed with the following warning:
```
Red Hat OpenShift Local requires at least 9GB of memory to run, with fewer memory only the microshift preset will be available
```

**Reply by praveenkumar** [reply]
@josecastillolema Yes, this is as now not wired with installer :( , we should be handling it separately once we start moving from POC to deliverables.

**Comment by josecastillolema** [comment]
There seems to be a problem with user configuration, i.e.:
```
% oc project kube-system
error: unable to default to a user name: the server could not find the requested resource (get users.user.openshift.io ~)
```

The **eval** option for loading k8s credentials is not working:
```
% eval $(crc oc-env)
% oc get node
error: Missing or incomplete configuration info.
```

So I exported the kubeconfig:
```
% export KUBECONFIG=/Users/jlema/.crc/machines/crc/kubeconfig
% oc get node
NAME              STATUS   ROLES                         AGE   VERSION
api.crc.testing   Ready    control-plane,master,worker   54s   v1.25.0
```

**Reply by praveenkumar** [reply]
@josecastillolema As of now microshift is bare minimal and doesn't have `user` or `project` resource so only way to interact with cluster is using the kubeconfig files.

**Reply by josecastillolema** [reply]
Hi @praveenkumar I have microshift installed on a RHEL 8 machine and I can change between projects, I think this issue is specific to the preset:
```
# systemctl status microshift
● microshift.service - MicroShift
   Loaded: loaded (/usr/lib/systemd/system/microshift.service; enabled; vendor >
   Active: active (running) since Mon 2022-12-05 04:53:56 EST; 2min 14s ago
 Main PID: 8574 (microshift)
    Tasks: 14 (limit: 11198)
   Memory: 562.1M
   CGroup: /system.slice/microshift.service
           └─8574 /usr/bin/microshift run

# oc project kube-system
Now using project "kube-system" on server "https://127.0.0.1:6443".
```

**Reply by praveenkumar** [reply]
@josecastillolema which version of the microshift are you using?

**Reply by josecastillolema** [reply]
```
# rpm -aq | grep microshift
microshift-selinux-4.8.0-2022_04_20_141053.el8.noarch
microshift-4.8.0-2022_04_20_141053.el8.x86_64
```

**Reply by praveenkumar** [reply]
@josecastillolema This is very old version of microshift and after refactor and making it productize doesn't provide those API's. https://github.com/openshift/microshift/blob/main/docs/getting_started.md as of now this is the starting point.

**Reply by josecastillolema** [reply]
Ack, thanks!
